ui/pages/settings/flask/snaps-list-tab/view-snap/view-snap.container.js

```diff
@@ -0,0 +1,38 @@
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { getSubjectMetadata } from '../../../../../selectors';
+import { removePermittedAccount } from '../../../../../store/actions';
+import ViewSnap from './view-snap.component';
+
+const mapStateToProps = (state, ownProps) => {
+  const { snap } = ownProps;
+  const subjectMetadata = getSubjectMetadata(state);
+  const connectedSubjects = Object.entries(subjectMetadata).reduce(
+    (val, [_, currSnap]) => {
+      if (currSnap.subjectType === 'snap') {
+        const { iconUrl, name, origin } = currSnap;
+        return [...val, { iconUrl, name, origin }];
+      }
+      return val;
+    },
+    [],
+  );
+  return {
+    snap,
+    connectedSubjects,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    onDisconnect: (origin, address) => {
+      dispatch(removePermittedAccount(origin, address));
+    },
+  };
+};
```

Comment:
This is using the wrong action. We will need to remove the Snap permission from the origin in question, the user's address has nothing to do with it.

Reply:
We need to use the `removePermissionsFor` action. Something like this:

```suggestion
const mapDispatchToProps = (dispatch) => {
  return {
    onDisconnect: (connectedOrigin, snapPermissionName) => {
      dispatch(removePermissionsFor({ [connectedOrigin]: [snapPermissionName] }));
    },
  };
};
```